blog/_posts/2025-07-15-network-perf-aks.md

```diff
@@ -0,0 +1,89 @@
+
+We compared the test results of Azure's older generation series [Dsv3](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes/general-purpose/dsv3-series?tabs=sizebasic) and newer generation series [Dsv6](https://learn.microsoft.com/en-us/azure/virtual-machines/sizes/general-purpose/dsv6-series?tabs=sizebasic) on AKS, as well as AWS next-generation instances [M7i](https://aws.amazon.com/ec2/instance-types/m7i/) on EKS, and observated signifncant network performance difference:
+
+Up to 35% higher throughput for Azure Dsv6 compared to Dsv3 and AWS M7i when trying to maximize network bandwidth usage.
+
+![image](/assets/images/network-perf-aks/sku_throughput.png)
+
+Up to 3x~6x lower RTT for Azure Dsv6 compared to Dsv3 and AWS M7i when tests are limited to the same network badnwdith usage.
+
+![image](/assets/images/network-perf-aks/sku_rtt.png)
+
+TCP retransmissions remained consistently at 0% on Azure Dsv6, matching AWS M7i while significantly outperforming Azure Dsv3.
+
+![image](/assets/images/network-perf-aks/sku_retransmits.png)
+
+The primary reasons for the significant leap in network performance on Dsv6 is because the next-generation network interface [Microsoft Azure Network Adaptor (MANA)](https://learn.microsoft.com/en-us/azure/virtual-network/accelerated-networking-mana-overview) support [Jumbo Frame](https://en.wikipedia.org/wiki/Jumbo_frame) and [MTU 9000](https://learn.microsoft.com/en-us/azure/virtual-network/how-to-virtual-machine-mtu?tabs=linux).
+
+A higher MTU (Maximum Transmission Unit) and MSS (Maximum Segment Size) allow TCP traffic to transmit more data per packet, reducing the total number of packets that need to be processed by network interface and OS kernenl. This leads to fewer hardware interrupts, less buffering, and reduced overhead in data movement, ultimately improving overall network efficiency.
+
+![image](/assets/images/network-perf-aks/sku_mtu.png)
+
+We realized there is no way to fully abstract hardware from the application — fundamentally, application network performance is dictated by the capabilities of the underlying CPU, memory, and network interfaces on the physical host. Identifying the appropriate VM SKU and series is essential to ensure the application meets its networking performance requirements.
+
+![image](/assets/images/network-perf-aks/sku_cpu_usage.png)
+
+The MTU on Azure Dsv6-series VMs is not set to 9000 by default. You can manually adjust the MTU for a specific network interface using the following Linux command:
+
+```bash
+ip link set $device mtu 9000
+```
+
+To enable Jumbo Frames across all nodes in an AKS cluster, you can deploy a DaemonSet, such as [this example](https://github.com/Azure/telescope/blob/main/modules/kustomize/mtu/overlays/azure/patch.yaml). In addition, you can optimize MTU settings dynamically using [Path MTU Discovery](https://learn.microsoft.com/en-us/azure/virtual-network/how-to-virtual-machine-mtu?tabs=linux#path-mtu-discovery)
```

Comment:
are there cons to this (that make it "off" by default)?

Reply:
yes, there will be fragramentation issue if it's enabled on VM SKUs with lower MTU support